clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/services/LaunchTemplateService.java

```diff
@@ -143,37 +130,155 @@ public LaunchTemplate createLaunchTemplate(
         false);
   }
 
+  public LaunchTemplateVersion modifyLaunchTemplate(
+      NetflixAmazonCredentials credentials,
+      ModifyServerGroupLaunchTemplateDescription description,
+      LaunchTemplateVersion sourceLtVersion,
+      boolean shouldUseMixedInstancesPolicy) {
+
+    RequestLaunchTemplateData data =
+        buildLaunchTemplateDataForModify(
+            credentials,
+            new AsgConfiguration(),
+            description,
+            sourceLtVersion,
+            shouldUseMixedInstancesPolicy);
+    CreateLaunchTemplateVersionResult result =
+        ec2.createLaunchTemplateVersion(
+            new CreateLaunchTemplateVersionRequest()
+                .withLaunchTemplateId(sourceLtVersion.getLaunchTemplateId())
+                .withLaunchTemplateData(data));
+
+    log.info(
+        String.format(
+            "Created new launch template version %s for launch template ID %s",
+            result.getLaunchTemplateVersion().getVersionNumber(),
+            result.getLaunchTemplateVersion().getLaunchTemplateId()));
+
+    return result.getLaunchTemplateVersion();
+  }
+
+  /**
+   * Delete a launch template version. A new launch template when it is modified.
+   * https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DeleteLaunchTemplateVersions.html
+   *
+   * @param launchTemplateId launch template ID for the version to delete
+   * @param versionToDelete launch template version to delete
+   */
+  public void deleteLaunchTemplateVersion(String launchTemplateId, Long versionToDelete) {
+    log.info(
+        String.format(
+            "Attempting to delete launch template version %s for launch template ID %s.",
+            versionToDelete, launchTemplateId));
+
+    DeleteLaunchTemplateVersionsResult result =
+        ec2.deleteLaunchTemplateVersions(
+            new DeleteLaunchTemplateVersionsRequest()
+                .withLaunchTemplateId(launchTemplateId)
+                .withVersions(String.valueOf(versionToDelete)));
+
+    if (result.getUnsuccessfullyDeletedLaunchTemplateVersions() != null
+        && !result.getUnsuccessfullyDeletedLaunchTemplateVersions().isEmpty()) {
+      DeleteLaunchTemplateVersionsResponseErrorItem responseErrorItem =
+          result.getUnsuccessfullyDeletedLaunchTemplateVersions().get(0);
+      ResponseError failureResponseError = responseErrorItem.getResponseError();
+
+      // certain error codes can be considered success when they match the desired end state.
+      // this also acts as a safety net in retry scenarios.
+      List<String> codesConsideredSuccess =
```

Comment:
@jeyrschabu does this make sense or would you prefer to fail on every failure, irrespective of the error code?

Reply:
Makes sense!